benches/bench.rs

```diff
@@ -149,6 +149,84 @@ fn bench_add_large_with_string(c: &mut Criterion) {
     });
 }
 
+fn bench_set_multi_small_strings(c: &mut Criterion) {
+    let rt = Runtime::new().unwrap();
+
+    c.bench_function("set_multi_small_strings", |b| {
+        b.to_async(&rt).iter_custom(|iters| async move {
+            let mut client = setup_client().await;
+            let kv = vec![("key1", "value1"), ("key2", "value2"), ("key3", "value3")];
+            let start = std::time::Instant::now();
+            for _ in 0..iters {
+                let _ = client.set_multi(kv.clone(), None, None).await;
+            }
+            start.elapsed()
+        });
+    });
+}
+
+fn bench_set_multi_large_strings(c: &mut Criterion) {
+    let rt = Runtime::new().unwrap();
+
+    c.bench_function("set_multi_large_strings", |b| {
+        b.to_async(&rt).iter_custom(|iters| async move {
+            let mut client = setup_client().await;
+            let large_payload = "a".repeat(LARGE_PAYLOAD_SIZE);
+            let kv = vec![
+                ("key1", &large_payload),
+                ("key2", &large_payload),
+                ("key3", &large_payload),
+            ];
+            let start = std::time::Instant::now();
+            for _ in 0..iters {
+                let _ = client.set_multi(kv.clone(), None, None).await;
+            }
+            start.elapsed()
+        });
+    });
+}
+
+fn bench_set_multi_with_100_large_string_values(c: &mut Criterion) {
+    let rt = Runtime::new().unwrap();
+
+    c.bench_function("set_multi_large_strings", |b| {
+        b.to_async(&rt).iter_custom(|iters| async move {
+            let mut client = setup_client().await;
+
+            let large_payload = "a".repeat(LARGE_PAYLOAD_SIZE);
+
+            let mut keys = Vec::with_capacity(100);
+            let key_strings: Vec<String> = (0..100).map(|i| format!("key{}", i)).collect();
+            keys.extend(key_strings.iter().map(|s| s.as_str()));
+            let values = vec![large_payload.as_str(); 100];
+
+            let kv: Vec<(&str, &str)> = keys.into_iter().zip(values).collect();
+
+            let start = std::time::Instant::now();
+            for _ in 0..iters {
+                let _ = client.set_multi(kv.clone(), None, None).await;
```

Comment:
I haven't been able to work out how to avoid the `kv.clone()` here, which I think is severely impacting the performance of this benchmark and isn't indicative of how the function would perform under real workloads.